exporters/otlp/otlpgrpc/otlp_integration_test.go

```diff
@@ -200,12 +203,185 @@ func TestNewExporter_collectorConnectionDiesThenReconnectsWhenInRestMode(t *test
 	require.NoError(t, nmc.Stop())
 }
 
+func TestNewExporter_FailThreeTimesAndSucceedAfter(t *testing.T) {
```

Comment:
We need to validate the times for these tests will not cause false positive failures on our slow Windows and Mac OS CI testing systems. Have we done any measurements of this? If not, [I've created toy PRs with added workflows](https://github.com/MrAlias/opentelemetry-go/pull/153/files) that just run a single test many times to search for failures and I think we could do the same here.

Reply:
I didn't measure it, I tried to define sane delays that I think should be be a problem to the CI but If you can do it and report the result here would be awesome.

Reply:
Running this locally I've be able to produce some errors:

```
cd exporters/otlp/otlpgrpc && go test -failfast -count=10000 -run=TestExporterExportFailureAndRecoveryModes
go: downloading github.com/cenkalti/backoff/v4 v4.1.0
--- FAIL: TestExporterExportFailureAndRecoveryModes (0.16s)
    --- FAIL: TestExporterExportFailureAndRecoveryModes/Retry_stops_if_takes_too_long (0.00s)
        mock_collector_test.go:219: Failed to get an endpoint: listen tcp :0: socket: too many open files
```

```
go test -failfast -count=10000 -run=TestExporterExportFailureAndRecoveryModes
--- FAIL: TestExporterExportFailureAndRecoveryModes (10.25s)        
    --- FAIL: TestExporterExportFailureAndRecoveryModes/Disabled_retry (10.00s)
        otlp_integration_test.go:374: 
            	Error Trace:	otlp_integration_test.go:374
            	            				otlp_integration_test.go:402
            	Error:      	Not equal: 
            	            	expected: "rpc error: code = Unavailable desc = unavailable"
            	            	actual  : "rpc error: code = DeadlineExceeded desc = context deadline exceeded"
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -1 +1 @@
            	            	-rpc error: code = Unavailable desc = unavailable
            	            	+rpc error: code = DeadlineExceeded desc = context deadline exceeded
            	Test:       	TestExporterExportFailureAndRecoveryModes/Disabled_retry
FAIL
```

It seems like this error is coming from the fact that we are not shutting down the mock collector.

Reply:
Here's the CI test runner I setup: https://github.com/MrAlias/opentelemetry-go/pull/154/checks?check_run_id=2421187590

Reply:
This looks to be resolved: https://github.com/MrAlias/opentelemetry-go/pull/154/checks?check_run_id=2452200642